### PR TITLE
feat: refactor UX for lacework generate cloud-account aws

### PR DESCRIPTION
**Pull request by PengyuanZhao**
## Summary

- Refactor the UX of `lacework generate cloud-account aws` according to the [proposal](https://docs.google.com/document/d/1-sStyx3FoNkGw79ywUTEIGgzM7orX2wLrBn7ELywT1w/edit#heading=h.w5ighgm8d8px).

## How did you test this change?

`make test`
`make integration-generation`

https://github.com/lacework/go-sdk/assets/7945666/5932d026-7781-43d7-a0a5-da128991bcd4

## Issue

https://lacework.atlassian.net/browse/GROW-2563

### CODE REVIEW

File: cli/cmd/generate_aws.go [hunk 1]
```diff
@@ -16,89 +17,114 @@ import (
...
+	QuestionCloudtrailExistingIamRoleExtID = "External ID for the existing IAM role:"
+
+	// Custom location Question
+	QuestionAwsOutputLocation = "Custom output location (optional):"
+
+	// Other options
+	AwsAdvancedOptDone = "Done" // Used in aws controltower and eks_audit
+
+	// Question labels
+	IconAgentless  = "[Agentless]"
```
Comment (dmurray-lacework) [inline cli/cmd/generate_aws.go:106]: nit: could these be a const?

File: cli/cmd/generate_aws.go [hunk 2]
```diff
@@ -16,89 +17,114 @@ import (
...
+	IconConfig     = "[Configuration]"
+	IconCloudTrail = "[CloudTrail]"
```
Comment (dmurray-lacework) [inline cli/cmd/generate_aws.go:108]: nit: Could these be const?

File: cli/cmd/generate_aws.go [hunk 3]
```diff
@@ -595,677 +671,580 @@ func validateAwsAssumeRole(val interface{}) error {
...
+			Opts:     []survey.AskOpt{survey.WithValidator(validateAwsArnFormat)},
+			Required: true,
 		},
 		{
-			Prompt:   &survey.Input{Message: QuestionExistingIamRoleExtID, Default: config.ExistingIamRole.ExternalId},
+			Icon:     IconCloudTrail,
+			Prompt:   &survey.Input{Message: QuestionCloudtrailExistingIamRoleExtID, Default: config.ExistingIamRole.ExternalId},
 			Response: &config.ExistingIamRole.ExternalId,
-			Opts:     []survey.AskOpt{survey.WithValidator(survey.Required)},
+			// Opts:     []survey.AskOpt{survey.WithValidator(survey.Required)},
```
Comment (dmurray-lacework) [inline cli/cmd/generate_aws.go:1080]: nit: does this commented out code need to be here

File: integration/aws_generation_test.go [hunk 4]
```diff
@@ -370,42 +438,45 @@ func TestGenerationAwsAdvancedOptsConsolidatedWithSubAccounts(t *testing.T) {
...
+		aws.WithCloudtrailUseExistingS3(true),
+		aws.WithCloudtrailName(cloudtrailName),
+		aws.WithExistingCloudtrailBucketArn(s3BucketArn),
 	).Generate()
 	assert.Equal(t, buildTf, tfResult)
 }
 
-// Test using Config with subaccounts
-func TestGenerationAwsAdvancedOptsConfigWithSubAccounts(t *testing.T) {
+// Test CloutTrail integration with new S3 bucket
```
Comment (dmurray-lacework) [inline integration/aws_generation_test.go:451]:
```suggestion
// Test CloudTrail integration with new S3 bucket
```

File: integration/aws_generation_test.go [hunk 5]
```diff
@@ -570,56 +641,44 @@ func TestGenerationAwsAdvancedOptsUseExistingElements(t *testing.T) {
...
+		aws.WithSqsEncryptionKeyArn(kmsArn),
 		aws.WithSqsEncryptionEnabled(true),
-		aws.WithSqsQueueName(queueName),
 	).Generate()
 	assert.Equal(t, buildTf, tfResult)
 }
 
-// Test use of creating S3 Bucket, SNS topic and SQS Queue
-func TestGenerationAwsAdvancedOptsCreateNewElements(t *testing.T) {
+// Test CloutTrail integration with existing IAM role
```
Comment (dmurray-lacework) [inline integration/aws_generation_test.go:654]:
```suggestion
// Test CloudTrail integration with existing IAM role
```

**Review by dmurray-lacework — APPROVED** [review]

File: cli/cmd/generate_aws.go (hunk 3 above)
Comment (PengyuanZhao) [reply cli/cmd/generate_aws.go]: Removed.

**Review by ipcrm — COMMENTED** [review]
A couple of tiny changes

**Pull request merged**